File: react-frontend/src/helpers/vpnStatus.ts
export const VPN_STATUS = {
    FAILED: "failed",
    PENDING: "pending",
    RUNNING: "running",
    TERMINATED: "terminated",
} as const;

export type VPNStatus = typeof VPN_STATUS[keyof typeof VPN_STATUS];

const VPN_STATUS_VALUES = Object.values(VPN_STATUS) as VPNStatus[];

export const normalizeVPNStatus = (status: unknown): VPNStatus | null => {
    if (typeof status !== "string") {
        return null;
    }

    const normalizedStatus = status.trim().toLowerCase();
    return VPN_STATUS_VALUES.includes(normalizedStatus as VPNStatus)
        ? normalizedStatus as VPNStatus
        : null;
};

export const formatVPNStatus = (status: VPNStatus) => {
    return status[0].toUpperCase() + status.slice(1);
};

File: lambda/Deploy/vpn_status.py
from enum import Enum


class VPNStatus(str, Enum):
    FAILED = "failed"
    PENDING = "pending"
    RUNNING = "running"
    TERMINATED = "terminated"


ACTIVE_VPN_STATUSES = [VPNStatus.PENDING.value, VPNStatus.RUNNING.value]


def normalize_vpn_status(status) -> VPNStatus | None:
    if status is None:
        return None
    if isinstance(status, VPNStatus):
        return status

    normalized_status = str(status).strip().lower()
    try:
        return VPNStatus(normalized_status)
    except ValueError:
        return None
